/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

message DevicePortDescription {
  required string uri = 1;
  enum Type {
    MIDI = 1;
  }
  required Type type = 2;

  optional string display_name = 3;
  optional bool readable = 4;
  optional bool writable = 5;
}

message DeviceDescription {
  required string uri = 1;
  enum Type {
    MIDI_CONTROLLER = 1;
  }
  required Type type = 2;

  optional string display_name = 3;
  repeated DevicePortDescription ports = 4;
}
